packages/x-components/src/components/modals/main-modal.vue

```diff
@@ -25,45 +23,57 @@
    *
    * @public
    */
-  @Component({
+  export default defineComponent({
     components: {
       BaseEventsModal
+    },
+    props: {
+      /**
+       * Animation to use for opening/closing the modal.
+       *
+       * @public
+       */
+      animation: {
+        type: [Vue, String]
```

Comment:
Create a new file in `x-components/src/utils/options-api.ts` and export there the value we'll use for the animation props across the project. Something like 
````typescript
export const animationProp = [String, Object, Function] as PropType<[DefineComponent | string]>;
````

Then use it here.